src/backend/joanie/signature/backends/lex_persona/client.py

```diff
@@ -0,0 +1,400 @@
+        )
+
+    def upload_file(self, *args, **kwargs):
+        """
+        Wrapper to start a signature procedure with 2 signers.
+        This method implies : to prepare a workflow, to upload
+        a document, and start the process to sign them.
+        """
+        self._create_workflow()
+        self._upload_file_to_workflow()
+        self._update_workflowstatus_to_start()
+        return self.workflow_id
+
+    def create_invitation_link(self, recipient_email: str, *args, **kwargs):
+        """
+        Create invitation link to sign the document for 1 email address.
+        It will bring the user to sign every document that are linked to his email address.
+        """
+        return self._create_invitation_link_to_sign(recipient_email=recipient_email)
+
+    def sign_selection_of_document(self, recipient_email: str, workflow_ids: list):
+        """
+        Create the link to go sign directly the selected documents by their workflow id.
+        """
+        token_jwt = self._get_jwt_token_from_invite_url(recipient_email=recipient_email)
+        if not token_jwt:
+            return "Token can not be extracted from the URL invite link"
+
+        response = self._sign_specific_selection_of_workflow(
+            workflow_ids=workflow_ids, token_jwt=token_jwt
+        )
+        return {"inviteUrlSelection": response["consentPageUrl"]}
+
+    def refuse_signature(self, recipient_email: str, workflow_ids: list):
+        """
+        Refuse the invite of a workflow.
+        """
+        token_jwt = self._get_jwt_token_from_invite_url(recipient_email=recipient_email)
+        if not token_jwt:
+            return "Token can not be extracted from the URL invite link"
+
+        for workflow_id in workflow_ids:
+            url = f"{self.base_url}/api/workflows/{workflow_id}/refuse"
+            headers = {
+                "Authorization": f"Bearer {token_jwt}",
+                "Content-Type": "application/json",
+            }
+            payload = {"reason": "I disagree."}
+            response = requests.post(url, headers=headers, json=payload, timeout=3)
+            return response.json()
+
+    def download_document(self, workflow_id: str):
+        """
+        Download the document of a specific workflow in PDF format.
+        """
+        url = f"{self.base_url}/api/workflows/{workflow_id}/downloadDocuments"
+        headers = {
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/json",
+        }
+        try:
+            response = requests.get(url, headers=headers, timeout=3)
+            if response.status_code == 200:
+                filename = "downloaded_template.pdf"
+                with open(filename, "wb") as pdf_file:
+                    pdf_file.write(response.content)
+                    return f"{filename}"
+            if response.status_code != 200:
+                return "Can not download document, please check your %s", workflow_id
```

Comment:
Raising this message as an exception might be better, because currently the caller function has no way to know that the string returned is not a filename (except matching it with "Can not download..." or trying to open the file)

Reply:
Do we actually need to send to the frontend the file in PDF format ? Or should we rather ask the signature provider if we can have an endpoint that could generate for us the link to donwload on their webpage the document ?

Reply:
After discussion on the 13th of September, we actually need a "one-time" download link for the signature provider that we will generate on demand. 
More news to come ...